scanpy/tests/test_clustering.py

```diff
@@ -8,7 +8,10 @@ def adata_neighbors():
 
 
 def test_leiden_basic(adata_neighbors):
-    sc.tl.leiden(adata_neighbors)
+    sc.tl.leiden(adata_neighbors, use_weights=True, directed=True)
+    sc.tl.leiden(adata_neighbors, use_weights=True, directed=False)
+    sc.tl.leiden(adata_neighbors, use_weights=False, directed=True)
+    sc.tl.leiden(adata_neighbors, use_weights=False, directed=False)
```

Comment:
Would you mind timing these tests vs what they were previously? I think I had previously done something like this, but it turned out the clustering took a while. Ditto for similar tests below.

Since we don't actually validate the values, these could be sped up by reducing the samples (possibly also sample classes) present.

Reply:
Around 188ms.

![image](https://user-images.githubusercontent.com/1140359/64661392-63c70d00-d412-11e9-89c9-9793478a16e6.png)

I think higher coverage is always better than increasing the test time unless the increase is really wild.

Reply:
Previously:

![image](https://user-images.githubusercontent.com/1140359/64661498-e18b1880-d412-11e9-934a-1c438e629743.png)

Reply:
For louvain:

![image](https://user-images.githubusercontent.com/1140359/64661560-2fa01c00-d413-11e9-8f81-fb59c3ced189.png)

Reply:
More coverage is always better, but I run the tests more often if it doesn’t take forever. I have been thinking we should have a separate more rigorous set of tests where execution time doesn’t matter, but that’s a separate topic.

These time changes are fine though. I think I had previously set `use_weights` and `directed` via fixtures, which caused the adata_neighbors fixture to get recalculated for each computation, which was the expensive part.